### PR TITLE
Refactor WebApp components

### PR DESCRIPTION
**Pull request by spiredevang**

### CODE REVIEW

File: Applications/WebPortal/WebApp/library/source/components/button.tsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ export class Button extends React.Component<Properties> {
       } else {
         return Button.STYLE.default;
       }
-    })();
+    })() as React.CSSProperties;
```
Comment (spiresheep) [inline Applications/WebPortal/WebApp/library/source/components/button.tsx:35]: There is no need to put the type assertion here.

File: Applications/WebPortal/WebApp/library/source/components/currency_selection_field.tsx [hunk 2]
```diff
@@ -41,7 +40,7 @@ export class CurrencySelectionField extends React.Component<Properties> {
       } else {
         return CurrencySelectionField.STYLE.selectionBox;
       }
-    })();
+    })() as React.CSSProperties;
```
Comment (spiresheep) [inline Applications/WebPortal/WebApp/library/source/components/currency_selection_field.tsx:43]: No need for type assertion after IIFE.

File: Applications/WebPortal/WebApp/library/source/components/date_time_field.tsx [hunk 3]
```diff
@@ -148,7 +145,7 @@ export class DateTimeField extends React.Component<Properties, State> {
     }
   }
 
-  private onPeriodChange(event: React.ChangeEvent<HTMLSelectElement>): void {
+  private onPeriodChange = (event: React.ChangeEvent<HTMLSelectElement>): void => {
```
Comment (spiresheep) [inline Applications/WebPortal/WebApp/library/source/components/date_time_field.tsx:148]: This doesn't need a return type anymore. Remove it.

File: Applications/WebPortal/WebApp/library/source/components/button.tsx [hunk 4]
```diff
@@ -56,7 +56,7 @@ export class Button extends React.Component<Properties> {
       outline: 'none',
       MozAppearance: 'none' as 'none',
       cursor: 'pointer' as 'pointer'
-    },
```
Comment (kranar) [inline Applications/WebPortal/WebApp/library/source/components/button.tsx:59]: One of the reasons for casting to ```as React.CSSProperties``` is to remove many of the ```'none' as 'none'```, ```'pointer' as 'pointer'```, and instead write simply ```'none'```, ```'pointer'```, so go ahead and remove the extraneous casts.

**Pull request merged**